qemu/tests/boot_from_virtiofs.py

```diff
@@ -0,0 +1,80 @@
+import os
+
+from avocado.utils import process
+from virttest import data_dir
+from virttest import qemu_vm
+
+
+def create_kernel_initrd(test, params):
+    """
+    Create initramfs and kernel file with dracut modul
+    """
+    install_path = data_dir.get_data_dir()
+    kernel_version = process.getoutput(params.get("guest_ver_cmd", "uname -r"))
+    create_initramfs_cmd = (params["create_initramfs_cmd"] % install_path)
+    status, output = process.getstatusoutput(create_initramfs_cmd)
+    if status:
+        test.fail("Failed to create initramfs.")
+    test.log.info("initramfs is created in %s" % install_path)
+    initrd_path = install_path + "/initramfs-virtiofs.img"
+
+    # copy vmlinuz to virtiofs_root
+    process.system("cp /boot/vmlinuz-%s %s" % (kernel_version, install_path))
+    kernel_path = install_path + ("/vmlinuz-%s" % kernel_version)
+    params["kernel"] = kernel_path
+    params["initrd"] = initrd_path
+    return kernel_path, initrd_path
+
+
+def setup_basic_root_fs(test, params):
+    """
+    create basic root file system
+    """
+    virtiofs_root_path = data_dir.get_data_dir() + "/virtiofs_root"
+    install_file_system_cmd = (params["install_file_system_cmd"]
+                               % virtiofs_root_path)
+    status, output = process.getstatusoutput(install_file_system_cmd)
+    if status:
+        test.fail("Failed to install basic root file system.")
+    return virtiofs_root_path
+
+
+def change_passwd(test, params, virtiofs_root_path):
+    """
+    change the password
+    """
+
+    check_selinux_cmd = params["check_selinux_cmd"]
+    selinux_value = process.getoutput(check_selinux_cmd)
+    if selinux_value != "Disabled":
+        process.getoutput("setenforce 0")
+
+    set_passwd_cmd = params["set_passwd_cmd"]
+    # change root path
+    os.chroot(virtiofs_root_path)
+    # change password
+    os.system(set_passwd_cmd)
+
+
+def boot_from_virtiofs(test, params, env):
+    """
+    generate qemu cmd line and start the vm
+    """
+    vm_name = "virtiofs_vm"
+    vm = qemu_vm.VM(vm_name, params, None, env['address_cache'])
+    env.register_vm(vm_name, vm)
+    vm.create()
+
+
+def run(test, params, env):
+    """
+    Boot vm from virtiofs test
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+    virtiofs_root_path = setup_basic_root_fs(test, params)
+    kernel_path, initrd_path = create_kernel_initrd(test, params)
+    change_passwd(test, params, virtiofs_root_path)
+    boot_from_virtiofs(test, params, env)
```

Comment:
And the initramfs files, or please use temp dir instead of data dir, or at least create a specific folder for this test under data direction. And make sure the files are cleaned even when the case failed.

Reply:
Yes,it is also in my plan,I will add a function to clean the env.